lib/prosopite.rb

```diff
@@ -187,16 +187,16 @@ def mysql_fingerprint(query)
       query
     end
 
-    def send_notifications
-      @custom_logger ||= false
-      @rails_logger ||= false
-      @stderr_logger ||= false
-      @prosopite_logger ||= false
-      @raise ||= false
-
-      notifications_str = ''
+    def clean_notifications(notifications)
+      notifications.map do |queries, kaller|
+        kaller = backtrace_cleaner.clean(kaller)
+        [queries, kaller]
+      end.to_h
+    end
 
-      tc[:prosopite_notifications].each do |queries, kaller|
+    def generate_notification_message(notifications)
+      notifications_str = StringIO.new
```

Comment:
Αfaik StringIO is useful for emulating the interface of writing/reading to files which is not the case here.